[Issue] transient failure during release pipeline run

- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `release pipeline`** 
```
Error Trace:          framework_service_docker_test.go:73 
Error:                   Not equal: 
                            expected: "Building docker image" 
                            actual  : "" 

                            Diff: 
                            --- Expected 
                            +++ Actual 
                            @@ -1 +1 @@ 
                            -Building docker image 
                            + 
Test: TestDefaultDockerOptions 
--- FAIL: TestDefaultDockerOptions (0.02s)
```

**Describe the bug**
The pipeline run should not fail since PR passed all tests and merged. Error traced from `TestDefaultDockerOptions` under `framework_service_docker_test.go`.

**To Reproduce**
Manually run [release pipeline](https://dev.azure.com/azure-sdk/internal/_build?definitionId=4643) on main.

**Expected behavior**
Could someone investigate this transient failure to validate if this is not hiding larger issues?

**Environment**
Information on your environment:
    * Language name and version
    * IDE and version : [e.g. Visual Studio 16.3]

--- a/cli/azd/pkg/project/framework_service_docker_test.go
+++ b/cli/azd/pkg/project/framework_service_docker_test.go
@@ -54,23 +54,27 @@ services:
 	docker := tools.NewDocker(dockerArgs)
 
 	progress := make(chan string)
-	defer close(progress)
+	done := make(chan struct{})
 
 	internalFramework := NewNpmProject(service.Config, &env)
-	status := ""
+	progressMessages := []string{}
 
 	go func() {
 		for value := range progress {
-			status = value
+			progressMessages = append(progressMessages, value)
 		}
+		done <- struct{}{}
 	}()
 
 	framework := NewDockerProject(service.Config, &env, docker, internalFramework)
 	res, err := framework.Package(ctx, progress)
+	close(progress)
+	<-done
 
 	require.Equal(t, "imageId", res)
 	require.Nil(t, err)
-	require.Equal(t, "Building docker image", status)
+	require.Len(t, progressMessages, 1)
+	require.Equal(t, "Building docker image", progressMessages[0])
 	require.Equal(t, true, ran)
 }
 

